packages/dashboard/src/components/robots/robot-decommission.tsx

```diff
@@ -0,0 +1,235 @@
+}: RobotDecommissionButtonProp): JSX.Element {
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+  const appController = React.useContext(AppControllerContext);
+  const [reassignTasks, setReassignTasks] = React.useState(true);
+  const [allowIdleBehavior, setAllowIdleBehavior] = React.useState(false);
+
+  const handleReassignTasksChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setReassignTasks(event.target.checked);
+  };
+
+  const handleAllowIdleBehaviorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAllowIdleBehavior(event.target.checked);
+  };
+
+  enum ConfirmDialogType {
+    None,
+    Decommission,
+    Recommission,
+  }
+  const [openConfirmDialog, setOpenConfirmDialog] = React.useState<ConfirmDialogType>(
+    ConfirmDialogType.None,
+  );
+
+  const robotDecommissioned =
+    robotState &&
+    robotState.commission &&
+    robotState.commission.dispatch_tasks === false &&
+    robotState.commission.direct_tasks === false;
+
+  // TODO: use authz for decommissioning robot
+  // const profile: UserProfile | null = React.useContext(UserProfileContext);
+  // const userCanDecommissionRobot = profile && Enforcer.canDecommissionRobot(profile);
+
+  const handleDecommission = React.useCallback<React.MouseEventHandler>(async () => {
+    if (!robotState || !robotState.name) {
+      return;
+    }
+    try {
+      if (!rmf) {
+        throw new Error('fleets api not available');
+      }
+      const resp = await rmf.fleetsApi?.decommissionRobotFleetsNameDecommissionPost(
+        fleet,
+        robotState.name,
+        reassignTasks,
+        allowIdleBehavior,
+      );
+
+      if (!resp.data.commission.success) {
+        appController.showAlert(
+          'error',
+          `Failed to decommission ${fleet}:${robotState.name}: ${
+            resp.data.commission.errors ?? ''
+          }`,
+        );
+      } else {
+        const errors = [];
+        if (
+          resp.data.pending_direct_tasks_policy &&
+          !resp.data.pending_direct_tasks_policy?.success
+        ) {
+          errors.push(`${resp.data.pending_direct_tasks_policy.errors}`);
+        }
+        if (
+          resp.data.pending_dispatch_tasks_policy &&
+          !resp.data.pending_dispatch_tasks_policy?.success
+        ) {
+          errors.push(`${resp.data.pending_dispatch_tasks_policy.errors}`);
+        }
+
+        appController.showAlert(
+          'success',
+          `Decommission of ${fleet}:${robotState.name} requested, ${
+            reassignTasks ? 'with' : 'without'
+          } task re-assignment, ${allowIdleBehavior ? 'allow' : 'not allowing'} idle behaviors${
+            errors.length !== 0 ? `, but with errors: ${errors}` : ''
+          }`,
+        );
+      }
+    } catch (e) {
+      appController.showAlert(
+        'error',
+        `Failed to decommission ${fleet}:${robotState.name}: ${(e as Error).message}`,
+      );
+    }
+    setReassignTasks(true);
```

Comment:
1. Refactor this into a function that reset the dialog.
2. Also reset `allowIdleBehavior`.